### DISCUSSION TITLE
NeighborSampler and accuracy computation in graph sage

### DISCUSSION THREAD

**Discussion by Laolga** [pyg-team/pytorch_geometric#3124] [Q&A] [answered]
Hi!
I've tried to use a similar architecture from this example https://github.com/pyg-team/pytorch_geometric/blob/c5152eae0de1bfaec1b60a7f931e0c928bb6df3c/examples/ogbn_products_sage.py#L27 as I also have a very big graph and a node classification task.  In the above example, the approximate training accuracy (which is computed based on subnetworks from train_loader) and the exact training accuracy (based on subgraph_loader where we take all nodes) are pretty much the same.  But in my case, the approximate accuracy looks ok (not too high but at least increasing while training loss is nicely decreasing)
![image](https://user-images.githubusercontent.com/33780205/133095364-145fcec8-3a0f-42ac-8367-704a9459f6cb.png)

And the exact accuracy is just pretty random:
![image](https://user-images.githubusercontent.com/33780205/133095465-9b82338c-226b-45ec-98b2-b0e1b6dc238b.png)
It's a binary classification and train_idx and test_idx are such that the classes are balanced. Any idea what I might be doing wrong? I've tried to increase/decrease subnetwork sizes in NeighborSampler but I don't see any difference.  I would also understand if train and test accuracy would diverge that much, but why train accuracy based on the train_loader is so different from train accuracy based on subgraph_loader?  Any idea why that might be happening? 
Right now I'm sampling like this:
```{python}
train_loader = NeighborSampler(data.edge_index, node_idx = train_idx,
                               sizes=[20, 20, 20], batch_size=1000,num_workers=20, shuffle = True)
subgraph_loader = NeighborSampler(data.edge_index, node_idx=None, sizes=[-1],
                                  batch_size=1000, shuffle=False,
                                  num_workers=20)
```

**Comment by rusty1s** [comment, marked as answer]
That's interesting. What happens if you evaluate with sampling Does that give you more reasonable performances? Other-wise, this might not be related to the usage of `subgraph_loader` at all, and may instead be due to some other issues, e.g., such as batch normalization doing some weird things.

**Comment by Laolga** [comment]
Well, that's awkward, but seems like I did not apply sigmoid to the inference function. Sorry for bothering and thanks for such a quick reply!

**Reply by rusty1s** [reply]
Awesome :)
